### ISSUE TITLE
Typo in `:help tag-priority`

### ISSUE THREAD

**Issue by jairlopez** [vim/vim#4509]
Hi,

I think I've found a little typo in the documentation displayed by [`:help tag-priority`](https://github.com/vim/vim/blob/4e0bf846279f3efa6299a98143033db1fdfa143a/runtime/doc/tagsrch.txt#L97); it currently says:
```
- when 'tagcase' is "followic" and 'ignorecase' is off
```
but it should say:
```
- when 'tagcase' is "followic" and 'ignorecase' is on
```
Thanks.

**Comment by jairlopez** [comment]
[Here](https://github.com/vim/vim/blob/260addf7955f3695d3daef9dcf840952af9fd851/runtime/doc/tagsrch.txt#L506) is another. It should say
```
This is a very good one. It works for C, C++, Java,
```
instead of
```
This a very good one. It works for C, C++, Java,
```

**Comment by brammool** [comment]
Hmm, these two lists don't look correct.  How about this:

The ignore-case matches are not found for a ":tag" command when:
- 'tagcase' is "followic" and the 'ignorecase' option is off
- 'tagcase' is "followscs" and the 'ignorecase' option is off and the
  'smartcase' option is off or the pattern contains an upper case character.
- 'tagcase' is "match"
- 'tagcase' is "smart" and the pattern contains an upper case character.

The ignore-case matches are found when:
- a pattern is used (starting with a "/")
- for ":tselect"
- when 'tagcase' is "followic" and 'ignorecase' is on
- when 'tagcase' is "followscs" and 'ignorecase' is on or the 'smartcase'
  option is on and the pattern does not contain an upper case character
- when 'tagcase' is "ignore"
- when 'tagcase' is "smart" and the patter does not contain an upper case
  character

**Comment by jairlopez** [comment]
That looks good, but [`:help
'smartcase'`](https://github.com/vim/vim/blob/4e0bf846279f3efa6299a98143033db1fdfa143a/runtime/doc/options.txt#L6808) already says it'll override the 'ignorecase' option only when the search pattern is typed and 'ignorecase' option is on, so I think the previous wording of: 
> - 'tagcase' is "followscs" and the 'ignorecase' option is off and the
> 'smartcase' option is off or the pattern contains an upper case character.

is just okay.

**Comment by chrisbra** [comment]
fixed by 12ee7ff00b91d852e060bb24951d1c94239
